R/num_equal.R

```diff
@@ -10,8 +10,13 @@ num_equal <- function(x, y, tolerance = default_tol()) {
     return(FALSE)
   }
 
-  attributes(x) <- NULL
-  attributes(y) <- NULL
+  if (is_int64(x) || is_int64(y)) {
+    x <- bit64::as.integer64(x)
```

Comment:
one concern here is trying to compare numbers from the nonoverlapping domains of int64 and double (forget the exact values off the top of my head)

i.e. an integer without exact representation as double vs a double too large for integer64

Reply:
Ooh yeah, good thought. 36893488147419103232  is an integer (2^65) that's you can represent with a double, but not in int64.  But of course not all integers bigger than that can be represented, e.g. `36893488147419103232 + 1 == 36893488147419103232` is TRUE.

Reply:
That too! I also had in mind the converse, integers near 2^60 that _can_ be represented in int64 but _not_ in double, for example any integer between `0x43B0000000000000 == 1152921504606846976` and `0x43B0000000000001==1152921504606847232` ([ref](https://numeral-systems.com/ieee-754-converter/))